File: C2/PDOO/Practicas/DeepSpaceRuby/lib/EnemyToUI.rb
module Deepspace

# 30.3.17 - Translation from Java
# @author Profesor
  
class EnemyToUI 
  
  attr_reader :name, :ammoPower, :shieldPower, :loot, :damage
    
  def initialize(enemy) 
    @name=enemy.name
    @ammoPower=enemy.ammoPower
    @shieldPower=enemy.shieldPower
    @loot=enemy.loot.getUIversion()
    @damage = enemy.damage.getUIversion()
  end

  def to_s
    return "Name: #{@name}, " + \
        "Ammo Power: #{@ammoPower}, " + \
        "Shield Power: #{@shieldPower}\n" + \
        "\tLoot..: #{@loot}\n" + \
        "\tDamage: #{@damage}"
  end
    
end # class

end # module

File: C2/PDOO/Practicas/DeepSpaceJava/src/deepspace/EnemyToUI.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author Profesor
 */
public class EnemyToUI {
    private final String name;
    private final float ammoPower;
    private final float shieldPower;
    
    private LootToUI loot;
    private DamageToUI damage;   
    
    EnemyToUI(EnemyStarShip enemy) {
        name=enemy.getName();
        ammoPower=enemy.getAmmoPower();
        shieldPower=enemy.getShieldPower();
    
        loot=enemy.getLoot().getUIversion();
        damage = enemy.getDamage().getUIversion();
    }

    public String getName() {
        return name;
    }

    public float getAmmoPower() {
        return ammoPower;
    }

    public float getShieldPower() {
        return shieldPower;
    }

    public LootToUI getLoot() {
        return loot;
    }

    public DamageToUI getDamage() {
        return damage;
    }
    
    
}
